api/v1beta1/foundationdbcluster_types.go

```diff
@@ -1783,7 +1811,24 @@ type FoundationDBBackupSpec struct {
 	// The cluster this backup is for.
 	ClusterName string `json:"clusterName"`
 
+	// +kubebuilder:validation:Enum=Running
+	// The desired state of the backup.
+	// The default is Running.
+	BackupState string `json:"backupState,omitempty"`
```

Comment:
Can we make this an enum instead of hardcoded strings for state? Especially once we add mode states such as stopped and paused.

Reply:
The closest we can get, I think, is to define a type called `BackupState` that is aliased to `string`, and then define string constants that represent the known states. This wouldn't restrict the possible values, though, and it would just turn the comparisons to strings we're making in the code into comparisons to constants for those strings. There's definitely some advantages there, but it's also more code, and I don't know if it's worth it. The other downside is that we would have to define the list of backup states both in that new list of constants and in the kubebuilder validations.

The number of states is going to stay small, and we're only going to be doing comparisons against them in the methods like `ShouldRun`, so I think having them stay as strings is going to be manageable.